src/dump1090.c

```diff
@@ -2538,57 +2538,77 @@ static void modeS_send_SBS_output (const modeS_message *mm, const aircraft *a)
 
   /* Field 11 could contain the call-sign we can get from `aircraft_find_or_create()::reg_num`.
    */
+
+  // Need to output the current date and time into the SBS output
+  //          1   2 3 4 5      6 7          8            9          10           11 12   13 14  15       16       17 18 19 20 21 22
+  // example: MSG,3,1,1,4CA7B6,1,2023/10/20,22:33:49.364,2023/10/20,22:33:49.403,  ,7250,  ,   ,53.26917,-2.17755,  ,  ,  ,  ,  ,0
+  time_t now = time(NULL);
+  char date_str[40];
+  struct tm* time_info = localtime(&now);
+
+  // Since the date,time,date,time is just a repeat, build the whole string once and then add it to each MSG output (note, had to remove 3 commas from the lines below)
+  strftime(date_str, sizeof date_str, "%Y/%m/%d,%H:%M:%S,%Y/%m/%d,%H:%M:%S", time_info);

```

Comment:
So there are 2 dates in this SBS output-format? Why and where is the spec. for this format?

Reply:
Wops, seems to be here:
http://woodair.net/sbs/article/barebones42_socket_data.htm

 * Field 7:  Date message generated 
 * Field 8:  Time message generated 
  * Field 9:  Date message logged 
  * Field 10:  Time message logged

Reply:
But your example shows the `%S` with milli-seconds. I do not find that `strftime()` supports that.
Besides letting this function parse all those formats seems terribly in-efficient. Would this be better:
```c
static const char *get_SBS_timestamp (void)
{
  FILETIME    ft;
  SYSTEMTIME  st;
  int         ts_len;
  char        ts_buf [20];
  static char timestamp [2*sizeof(ts_buf)];

  get_FILETIME_now (&ft);
  FileTimeToSystemTime (&ft, &st);
  ts_len = snprintf (ts_buf, sizeof(ts_buf), "%04u/%02u/%02u,%02u:%02u:%02u.%03u,",
                     st.wYear, st.wMonth, st.wDay, st.wHour, st.wMinute, st.wSecond, st.wMilliseconds);
  strcpy (timestamp, ts_buf);
  strcat (timestamp, ts_buf);
  timestamp [ts_len] = '\0';    /* remove last ',' */
  return (timestamp);
}
```

Reply:
Yea, glad you found the bit in the spec. It's the same spec I've been using for many years! :) I've never seen fields 7/9 and 8/10 differ so have made the assumption that they're just set the same.

Initially I thought about building up the string manually in a similar way, but when I looked at the template string, just calling strftime felt like it offloaded the work to that function which could probably do a more efficient job of it than I could (since it's a C++ function) and it simplified coding inside dump1090.c, but what you have there makes sense and gets milliseconds back into the string.